### PR TITLE
AS: affiliate fees

### PR DESCRIPTION
**Pull request by dorothy-zbornak**
## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/asset-swapper/src/types.ts [hunk 1]
```diff
@@ -124,13 +124,21 @@ export interface ForwarderExtensionContractOpts {
     feeRecipient: string;
 }
 
+export interface AffiliateFee {
+    recipient: string;
+    buyTokenFeeAmount: BigNumber;
```
Comment (dorothy-zbornak) [inline packages/asset-swapper/src/types.ts:129]: MM wants percentage fees.

File: packages/order-utils/src/transformer_data_encoders.ts [hunk 2]
```diff
@@ -147,17 +147,11 @@ export function decodePayTakerTransformerData(encoded: string): PayTakerTransfor
  */
 export const affiliateFeeTransformerDataEncoder = AbiEncoder.create({
```
Comment (dorothy-zbornak) [inline packages/order-utils/src/transformer_data_encoders.ts]: All this needs to be reverted to original. I goofed. `AbiEncoder` can't handle this kind of encoding for some reason and it ends up reverting.

File: packages/asset-swapper/src/types.ts (hunk 1 above)
Comment (dorothy-zbornak) [reply packages/asset-swapper/src/types.ts:129]: Or I guess you're gonna do the conversion in API?

Comment (moodlezoup) [reply packages/asset-swapper/src/types.ts:129]: I'm handling the percentages from the 0x-api side

**Review by dekz — COMMENTED** [review]
LGTM

**Pull request merged**